modules/reactor/src/meshgenerators/PatternedCartesianPeripheralModifier.C

```diff
@@ -0,0 +1,49 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PatternedCartesianPeripheralModifier.h"
+
+#include "MooseMeshUtils.h"
+
+registerMooseObject("ReactorApp", PatternedCartesianPeripheralModifier);
+
+InputParameters
+PatternedCartesianPeripheralModifier::validParams()
+{
+  InputParameters params = PatternedPolygonPeripheralModifierBase::validParams();
+  return params;
+}
+
+PatternedCartesianPeripheralModifier::PatternedCartesianPeripheralModifier(
+    const InputParameters & parameters)
+  : PatternedPolygonPeripheralModifierBase(parameters),
+    _square_peripheral_trimmability(
+        declareMeshProperty<bool>("square_peripheral_trimmability", false)),
+    _square_center_trimmability(declareMeshProperty<bool>("square_center_trimmability", false))
+{
+  _num_sides = SQUARE_NUM_SIDES;
+}
+
+std::unique_ptr<MeshBase>
+PatternedCartesianPeripheralModifier::generate()
+{
+  if (hasMeshProperty("square_center_trimmability", _input_name))
+    _square_center_trimmability = getMeshProperty<bool>("square_center_trimmability", _input_name);
+
+  // Check if the input mesh is compatible
+  if (!getMeshProperty<bool>("peripheral_modifier_compatible", _input_name))
+    paramError("input",
+               "The input mesh is generated by a mesh generator that is not compatible with "
+               "PatternedCartesianPeripheralModifier.");
+  // Whether the input mesh is cartesian
+  if (!hasMeshProperty("square_center_trimmability", _input_name))
+    paramError("input", "The input mesh is not declared as center-trimmable in its metadata.");
+
+  return PatternedPolygonPeripheralModifierBase::generate();
+}
```

Comment:
I used to get "no newline at EOF" failures?
@milljm did we change this?

Reply:
I think the empty line is there but not shown in github editor.